eng/targets/VisualStudio.targets

```diff
@@ -73,35 +78,27 @@
 
       <_AssetsByPackageId Remove="@(_ExcludedPackageId)" />
     </ItemGroup>
-    
-    <ItemGroup>
-      <_PackageToOptimize Include="@(NuGetPackageToIncludeInVsix)" Condition="'%(NuGetPackageToIncludeInVsix.Optimization)' == 'true'" />
-      <_PackageNotToOptimize Include="@(NuGetPackageToIncludeInVsix)" Condition="'%(NuGetPackageToIncludeInVsix.Optimization)' != 'true'" />
-
-      <_AssetsToOptimize Include="@(_AssetsByPackageId)" />
-      <_AssetsToOptimize Remove="@(_PackageNotToOptimize)" />
 
-      <_AssetsNotToOptimize Include="@(_AssetsByPackageId)" />
-      <_AssetsNotToOptimize Remove="@(_PackageToOptimize)" />
+    <!-- Flow metadata set on NuGetPackageToIncludeInVsix to VSIXCopyLocalReferenceSourceItem -->
+    <JoinItems Left="@(_AssetsByPackageId)" LeftKey="" LeftMetadata="*"
+               Right="@(NuGetPackageToIncludeInVsix)" RightKey="" RightMetadata="*">
+      <Output TaskParameter="JoinResult" ItemName="_AssetsWithMetadata" />
+    </JoinItems>
```

Comment:
What is this all doing here? We made _AssetsByPackageId, can't we just flow the extra property along there?

Reply:
_AssetsByPackageId originates from ReferenceCopyLocalPaths items, while we need to flow metadata from NuGetPackageToIncludeInVsix  items. So this is doing exactly that. It does a join operation on the two "tables".